eng/validation/build-arcadewithrepo.ps1

```diff
@@ -195,52 +210,105 @@ function Git-Command($repoName) {
-        Git-Command $targetRepoName push origin --delete $targetBranch
+        if($true -eq $pushBranchToGithub)
+        {
+            & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcGitHubRepoName --github-pat $githubPAT --password $barToken
+            Git-Command $githubRepoName push origin --delete $targetBranch
+        }
+        else
+        {
+            & darc delete-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcAzDORepoName --azdev-pat $azdoToken --password $barToken
+            Git-Command $githubRepoName remote add $remoteName $azdoUri
+            Git-Command $githubRepoName push $remoteName --delete $targetBranch
+        }
     }
     catch
     {
-        Write-Warning "Unable to delete default or branch when cleaning up existing branch"
+        Write-Warning "Unable to delete default channel or branch when cleaning up existing branch"
     }
 }
-Git-Command $targetRepoName checkout -b $targetBranch $sha
+
+## Create a branch from the repo with the given SHA.
+Git-Command $githubRepoName checkout -b $targetBranch $sha
 
 ## Make the changes to that branch to update Arcade - use darc
-Set-Location $(Get-Repo-Location $targetRepoName)
-& darc update-dependencies --channel ".NET Eng - Validation" --source-repo "arcade" --github-pat $githubPAT --password $barToken
+Set-Location $(Get-Repo-Location $githubRepoName)
+& darc update-dependencies --channel ".NET Eng - Validation" --source-repo "arcade" --github-pat $githubPAT --azdev-pat $azdoToken --password $barToken
 
-## Push branch to github
-Git-Command $targetRepoName commit -am "Arcade Validation test branch - version ${arcadeSdkVersion}"
-Git-Command $targetRepoName push origin HEAD
+Git-Command $githubRepoName commit -am "Arcade Validation test branch - version ${arcadeSdkVersion}"
 
-## Push branch to AzDO org/project with the official pipeline to build the repo
-## Don't need to do this for Roslyn, but we'll need to do it for Runtime. 
+if($true -eq $pushBranchToGithub)
+{
+    ## Push branch to github
+    Git-Command $githubRepoName push origin HEAD
 
-## Add default channel from that github repo and branch to "General Testing"
-& darc add-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcRepoName --github-pat $githubPAT --password $barToken
+    ## Add default channel from that github repo and branch to "General Testing"
+    & darc add-default-channel --channel "General Testing" --branch $darcBranchName --repo $darcGitHubRepoName --github-pat $githubPAT --password $barToken
+}
+else
+{
+    ## Push branch to AzDO org/project with the official pipeline to build the repo
+    ## make remote, it might already exist if we had to delete it earlier, so wrapping it in a try/catch
+    try
+    {
+        Git-Command $githubRepoName remote add $remoteName $azdoUri
+    }
+    catch
+    {
```

Comment:
won't this swallow legitimate errors? such as auth failures or azdo being down? Does it matter?

Reply:
Not for a remote add. There's no contact with external endpoints for that.